### DISCUSSION TITLE
How to configure noice for native cmdline ui?

### DISCUSSION THREAD

**Discussion by chuan137** [LazyVim/LazyVim#1926] [General]
I use lazyvim starter with my own customization. I want to get the native cmdline ui, shown at bottom. Following the docs, I disable both `cmdline` and `messages`. However the popup menu is not shown next to cmdline, rather next to the current cursor location. E.g. following screenshot when I type ':co' then tab in cmdline

<img width="765" alt="Screenshot 2023-11-01 at 13 52 09" src="https://github.com/LazyVim/LazyVim/assets/6780922/f30fdf7d-e40b-48be-a2c4-2567c6544de0">

This is my configuration in plugins/. I think the preset  command_palette is enabled by Lazyvim by default, therefore the popup menu should be at bottom, right?

```
return {
  { "rcarriga/nvim-notify", enabled = false },
  { "nvimdev/dashboard-nvim", enabled = false },

  -- classic cmdline
  {
    "folke/noice.nvim",
    opts = {
      cmdline = {
        enabled = false,
      },
      messages = {
        enabled = false,
      },
    },
  },
}
```

**Comment by dpetka2001** [comment]
It seems that adding
```lua
      popupmenu = {
        enabled = false,
      },
```
to your configuration does the trick? Test it out to see if it's what you want. Not sure if that's how it's supposed to work. Maybe it's a bug with `noice` if you want to keep the `popupmenu` enabled? Not sure. Maybe i'm entirely wrong about it being a bug cuz I don't understand noice so well myself.

**Reply by chuan137** [reply]
ty, it works. Is `popupmenu` only for the cmdline?

**Reply by dpetka2001** [reply]
Not sure. Haven't dabbled with noice configuration at all. Better to ask over at the repo maybe? I believe you'll get more precise answer.

**Comment by VimYoung** [comment]
The issue is quite old, but I am adding the following comment for future readers looking to disable the default hover command line of LazyVim, I found the following alternative more appropriate, it moves the cmdline to the bottom (like original) but overlaps with the lualine.
```
cmdline = {
    view = "cmdline",
},
```
 I liked it more as it saves a row unlike the traditional cmdline. To achieve the exact old one you have to do what is mentioned in above comments rather than this.
 Also disabling `popmenu` disables all the `noice` popmenu UI wherever it is use, not just the cmdline. So, keep that in mind before disabling if you are going for the old cmdline UI.

**Comment by elsoods** [comment]
Is there a way I could show command suggestions without popup menu? I keep trying to bring the suggestions menu to the bottom when using "cmdline" view= "cmdline" but it keeps showing up in the top as the default behaviour.

**Reply by matt7aylor** [reply]
This is what I am doing  to get something like a normal cmdline with completions:
(`~/.config/nvim/lua/plugins/noice.lua`)
```lua
return {
  {
    "folke/noice.nvim",
    -- Classic commandline position instead of pop-up
    opts = {
      cmdline = {
        view = "cmdline", -- moves command line to bottom
      },
      presets = { command_palette = false }, -- tab completions for commandline don't pop-up at top
    },
  },
}
```

**Reply by elsoods** [reply]
![Screenshot_20250423_013729](https://github.com/user-attachments/assets/a6dcb0e3-632c-4a1f-9f21-e19f4bab985f)

thanks! gave up a while ago on using noice. I was just tinkering with my config and stumbled upon this old comment i forgot about. That did it! thanks a lot.
